pczt/src/sapling.rs

```diff
@@ -0,0 +1,526 @@
+use std::collections::BTreeMap;
+
+use getset::MutGetters;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::{
+    common::Zip32Derivation,
+    roles::combiner::{merge_map, merge_optional},
+};
+
+const GROTH_PROOF_SIZE: usize = 48 + 96 + 48;
+
+/// PCZT fields that are specific to producing the transaction's Sapling bundle (if any).
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Bundle {
+    pub(crate) spends: Vec<Spend>,
+    pub(crate) outputs: Vec<Output>,
+
+    /// The net value of Sapling spends minus outputs.
+    ///
+    /// This is initialized by the Creator, and updated by the Constructor as spends or
+    /// outputs are added to the PCZT. It enables per-spend and per-output values to be
+    /// redacted from the PCZT after they are no longer necessary.
+    pub(crate) value_sum: i128,
+
+    /// The Sapling anchor for this transaction.
+    ///
+    /// Set by the Creator.
+    pub(crate) anchor: [u8; 32],
```

Comment:
Similar reasoning to Orchard:
```suggestion
    /// Set by the Creator or Constructor.
    pub(crate) anchor: Option<[u8; 32]>,
```